src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -311,6 +312,15 @@
    jruby-request/wrap-with-error-handling
    ring/wrap-params))
 
+(schema/defn ^:always-validate valid-static-file-path?
+  "Helper function to decide if a static_file_content path is valid.
+  The access here is designed to mimic Puppet's file_content endpoint."
+  [path :- schema/Str]
+  (when-let [canonicalized-path (URIUtil/canonicalPath path)]
+    (or
+     (bidi.bidi/match-route [["modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path)
+     (bidi.bidi/match-route [["environments/" :environment-name "/modules/" :module-name "/files/" [#".*" :rest]] :helper] canonicalized-path))))
+
```

Comment:
I can certainly add a comment explaining it and shorten the lines. I'm not sure how much simpler it can get. I guess extracting part of it into a function would probably work to simplify a little bit.